### PR TITLE
KeyBuilder: add datetime hashing

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: pytools/persistent_dict.py [hunk 1]
```diff
@@ -449,6 +449,9 @@ def update_for_frozendict(self, key_hash: Hash, key: Mapping) -> None:
     update_for_PMap = update_for_frozendict  # noqa: N815
     update_for_Map = update_for_frozendict  # noqa: N815
 
+    def update_for_datetime(self, key_hash: Hash, key: Any) -> None:
+        self.rec(key_hash, key.isoformat(timespec="microseconds"))
```
Comment (inducer) [inline pytools/persistent_dict.py:453]: I don't think this is correct. Two `datetime`s from different timezones can compare equal. Look at what `datetime.__eq__` does? Maybe use `timestamp`? (And please add a test for the timezone thing.)

**Comment by inducer** [comment]
Could you look at the failures here? (Or move this to draft status if it's not ready?)

**Comment by matthiasdiener** [comment]
Thanks @inducer , I think this is ready for another look

**Comment by inducer** [comment]
Thx!

**Pull request merged**